berkdb/btree/bt_cursor.c

```diff
@@ -3161,8 +3161,10 @@ split:	ret = stack = 0;
 		 */
 
         /* Invalidate the cursor before releasing the pagelock */
-		cp->pgno = PGNO_INVALID;
-		cp->indx = 0;
+		if (own == 0) {
+			cp->pgno = PGNO_INVALID;
+			cp->indx = 0;
+		}
```

Comment:
This change was discussed with @markhannum and `if (own == 0)` was what berkdb originally had. Also see Mark's last change [here](https://github.com/bloomberg/comdb2-old/pull/723/files#diff-276bb96720dfac5f10f48b22dca4cb37R3159).